core/app/models/spree/promotion.rb

```diff
@@ -112,13 +112,11 @@ def activate(order:, line_item: nil, user: nil, path: nil, promotion_code: nil)
       # If an action has been taken, report back to whatever activated this promotion.
       action_taken = results.include?(true)
 
-      if action_taken
-        # connect to the order
-        order_promotions.find_or_create_by!(
-          order_id: order.id,
-          promotion_code_id: promotion_code.try!(:id)
-        )
-      end
+      # connect to the order
+      order_promotions.find_or_create_by!(
+        order_id: order.id,
+        promotion_code_id: promotion_code.try!(:id)
+      )
 
       action_taken
```

Comment:
@jrochkind:

> Is anything in solidus using the return value?

promotion_handler/coupon.rb is the one place in Solidus that is - see [here](https://github.com/solidusio/solidus/pull/1428/files#diff-ac986bca0f198481ed44c93b87cd7c02R63).

> false for nothing happened (is that even still possible?)

It is still possible that nothing happened via [this](https://github.com/solidusio/solidus/blob/v1.3.1/core/app/models/spree/promotion.rb#L86), which would return nil. So we also have:

`nil` now means that the promotion was not attached.

I chatted with @jhawthorn about this offline and he wondered whether we shouldn't just have this return true when the promotion was attached and false when it wasn't.